lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -238,10 +238,18 @@ def given_error
           return " but nothing was raised" unless @actual_error
 
           backtrace = format_backtrace(@actual_error.backtrace)
+          if @actual_error.message.nil? || @actual_error.message.empty? || (@actual_error.message == @actual_error.class.to_s)
+            message_text = ''
+          else
+            message_lines = @actual_error.message.split(/\n/)
+            message_text = [" with message:", *message_lines].zip(["\n  # "] * message_lines.size)
+            message_text << "\nand"
+          end
           [
-            ", got #{description_of(@actual_error)} with backtrace:",
-            *backtrace
-          ].join("\n  # ")
+            ", got #{description_of(@actual_error)}",
+            message_text,
+            [" with backtrace:", *backtrace].zip(["\n  # "] * backtrace.size)
+          ].flatten.join('')
```

Comment:
This array could probably just be interpolation now, as the intent of the original map is taken care of by the zip.